EIPS/eip-2876.md

```diff
@@ -0,0 +1,188 @@
+
+The Deposit Address format can be generated with the following JavaScript code:
+
+```js
+/**
+ * Converts a 20 byte account address and a 5 byte nonce to a deposit address.
+ * The format of the return value is 28 bytes as follows. The + operator is byte
+ * concatenation.
+ * (baseAddress + nonce + keccak256(baseAddress + nonce)[:3])
+ *
+ * @param {String} baseAddress the given HEX address (20 byte hex string with 0x prepended)
+ * @param {String} nonce the given HEX nonce (5 byte hex string with 0x prepended)
+ * @return {String}
+ */
+function generateAddress (baseAddress, nonce) {
+  if (
+    !baseAddress.match(/^0x[0-9a-fA-F]{40}$/) ||
+    !nonce.match(/^0x[0-9a-fA-F]{10}$/)
+  ) {
+    throw new Error('Base Address and nonce must be 0x hex strings');
+  }
+  const ret =
+    baseAddress.toLowerCase() + nonce.toLowerCase().replace(/^0x/, '');
+  const myHash = web3.utils.keccak256(ret);
+  return ret + myHash.slice(2, 8); // first 3 bytes from the 0x hex string
+};
+```
+
+The checksum can be verified within the deposit contract itself using the following:
+
+```solidity
+function checksumMatch(bytes8 id) internal view returns (bool) {
+    bytes32 chkhash = keccak256(
+        abi.encodePacked(address(this), bytes5(id))
+    );
+    bytes3 chkh = bytes3(chkhash);
+    bytes3 chki = bytes3(bytes8(uint64(id) << 40));
+    return chkh == chki;
+}
+```
+
+### Part 2: The Contract Interface
+
+A contract that follows this ERC:
+
+- `The contract` MUST revert if sent a transaction where `msg.data` is null (A pure value transaction).
+- `The contract` MUST have a function a deposit function as follows:
+
+```solidity
+interface DepositEIP {
+  function deposit(bytes8 id) external payable returns (bool);
+}
+```
+
+- `deposit(bytes8)` MUST return `false` when the contract needs to keep the value, but signal to the depositor that the deposit (in terms of the parent application) itself has not yet succeeded. (This can be used for partial payment, ie. the invoice is for 5 ETH, sending 3 ETH returns false, but sending a second tx with 2 ETH will return true.)
+- `deposit(bytes8)` MUST revert if the deposit somehow failed and the contract does not need to keep the value sent.
+- `deposit(bytes8)` MUST return `true` if the value will be kept and the payment is logically considered complete by the parent application (exchange/merchant).
+- `deposit(bytes8)` SHOULD check the checksum contained within the 8 byte id. (See Part 1 for an example)
+- `The parent application` SHOULD return any excess value received if the deposit id is a one-time-use invoice that has a set value and the value received is higher than the set value. However, this SHOULD NOT be done by sending back to `msg.sender` directly, but rather should be noted in the parent application and the depositor should be contacted out-of-band to the best of the application manager's ability.
+
+### Part 3: Depositing Value to the Contract from a Wallet
```

Comment:
Consider removing the `Part #` bit off of these subsections.  The title is likely enough, and the order of implementation isn't important for the specification so it isn't necessary to enumerate the sections.